src/Microsoft.Identity.Web.Certificate/DefaultCredentialsLoader.cs

```diff
@@ -15,6 +16,9 @@ namespace Microsoft.Identity.Web
     public class DefaultCredentialsLoader : ICredentialsLoader
     {
         ILogger<DefaultCredentialsLoader>? _logger;
+        
+        private readonly Dictionary<string, SemaphoreSlim> _loadingSemaphores = new Dictionary<string, SemaphoreSlim>();
+        private readonly object _semaphoreLock = new object();
```

Comment:
can't we just do a double lock on the credentialDescription?

Reply:
Locks are easy to easy but will only allow one request to access a section of the code. Main reason for using a SephamoreSlim over a lock is that while we want only a single request to load a certificate for a specific CredentialDescription we also want to allow other requests to be able to access/load certificates for other CredentialDescriptions.

Reply:
Simplified the implementation using a ConcurrentDictionary and a SemaphoreSlim.